lib/tests/test_merge_trees.rs

```diff
@@ -627,13 +629,64 @@ fn test_simplify_conflict_after_resolving_parent() {
 // TODO: Add tests for simplification of multi-way conflicts. Both the content
 // and the executable bit need testing.
 
+#[test]
+fn test_rebase_lineariaze_lossy_merge() {
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    // Test this rebase:
+    // D    foo=2          D' foo=2
+    // |\                  |
+    // | C  foo=2          |
+    // | |           =>    B  foo=2
+    // B |  foo=2          |
+    // |/                  |
+    // A    foo=1          A  foo=1
+    //
+    // Commit D effectively discarded a change from "1" to "2", so one
```

Comment:
question: I didn't understand this. In what sense did this effectively discard a change from `1` to `2`? Isn't the value `2` in all parent commits? Why would `D` become `1` when rebased onto `B`?

Reply:
I've updated the description. Let me know if it's still unclear.